lib/vm.js

```diff
@@ -46,6 +46,18 @@ const {
 
 class Script extends ContextifyScript {
   constructor(code, options) {
+    if (options !== undefined && options.linker) {
+      if (typeof options.linker !== 'function') {
+        throw new ERR_INVALID_ARG_TYPE(
+          'options.linker', 'function', 'options.linker');
+      }
+      const linker = options.linker;
+      const { callLinkerForNamespace } = require('internal/vm/module');
+      options.importModuleDynamically = async (specifier) => {
+        return callLinkerForNamespace(linker, specifier, this);
```

Comment:
When `linker` is undefined, it seems like dynamic import is still throwing here.

It could be worthwhile still defaulting to the default Node loader here.

Reply:
the only way to get this far is if `options.linker` is a function, and i think defaulting to the process loader would be confusing/unexpected behavior

Reply:
Ah, thanks for clarifying, of course you've defined the error for this case as well - looks good.